### PR TITLE
[Merged by Bors] - chore: superfluous parentheses part 2

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mo271** [comment]
the ones in calc mode were suggested by @YaelDillies: https://github.com/leanprover-community/mathlib4/pull/12116#pullrequestreview-1999538171

File: Mathlib/Order/RelSeries.lean [hunk 1]
```diff
@@ -256,7 +256,7 @@ For two types `α, β` and relation on them `r, s`, if `f : α → β` preserves
 def map (p : RelSeries r) (f : r →r s) : RelSeries s where
   length := p.length
   toFun := f.1.comp p
-  step := (f.2 <| p.step .)
+  step := (f.2 <| p.step ·)
```
Comment (YaelDillies) [inline Mathlib/Order/RelSeries.lean:259]: Maybe should be a separate PR? I assume there are more than one

File: Mathlib/Tactic/ProxyType.lean [hunk 2]
```diff
@@ -229,7 +229,7 @@ def ensureProxyEquiv (config : ProxyEquivConfig) (indVal : InductiveVal) : TermE
           levelParams := indVal.levelParams
           safety := DefinitionSafety.safe
           hints := ReducibilityHints.abbrev
-          type := (← inferType equiv')
+          type := ← inferType equiv'
```
Comment (YaelDillies) [inline Mathlib/Tactic/ProxyType.lean:232]: Could @eric-wieser confirm that this is still correct?

Comment (eric-wieser) [reply Mathlib/Tactic/ProxyType.lean:232]: Yes, there is no other possible interpretation here other than a syntax error, which CI would catch.

File: Mathlib/Algebra/Module/Zlattice.lean [hunk 3]
```diff
@@ -549,7 +549,7 @@ theorem Zlattice.rank [hs : IsZlattice K L] : finrank ℤ L = finrank K E := by
     -- We prove finally that `e ∪ {v}` is not ℤ-linear independent or, equivalently,
     -- not ℚ-linear independent by showing that `v ∈ span ℚ e`.
     rw [LinearIndependent.iff_fractionRing ℤ ℚ,
-      (linearIndependent_insert (Set.not_mem_of_mem_diff hv)),  not_and, not_not]
+        linearIndependent_insert (Set.not_mem_of_mem_diff hv),  not_and, not_not]
```
Comment (mo271) [inline Mathlib/Algebra/Module/Zlattice.lean:552]:
```suggestion
      linearIndependent_insert (Set.not_mem_of_mem_diff hv),  not_and, not_not]
```

[the remark this answers is not in the document]
Comment (mo271) [reply Mathlib/Algebra/Module/Zlattice.lean]: fixed

File: Mathlib/Order/RelSeries.lean (hunk 1 above)
Comment (mo271) [reply Mathlib/Order/RelSeries.lean:259]: rolled back that change, will do a separate PR

Comment (mo271) [inline Mathlib/Order/RelSeries.lean:259]:
```suggestion
  step := (f.2 <| p.step .)
```

**Comment by YaelDillies** [comment]
Thanks!

maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge
